manager/allocator/network.go

```diff
@@ -1089,7 +1089,11 @@ func updateTaskStatus(t *api.Task, newStatus api.TaskState, message string) {
 
 // IsIngressNetwork returns whether the passed network is an ingress network.
 func IsIngressNetwork(nw *api.Network) bool {
-	if nw.Spec.Ingress {
+	cnmSpec := nw.Spec.GetCNM()
+	if cnmSpec == nil {
+		return false
+	}
```

Comment:
My thinking is that if `cnmSpec` (and all the other fields in `oneof spec`) are nil, we'd assume that the legacy fields are in use and honor those instead.